public/editor.js

```diff
@@ -1277,6 +1277,102 @@ class UINode
     }
 }
 
+class BitCrushNode extends UINode
+{
+    constructor(id, state, editor)
+    {
+        super(id, state, editor);
+
+        //~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+        // Bitdepth selection dropdown menu
+        let div = document.createElement('div');
+        div.className = 'node_center';
+
+        let label = document.createElement("label");
+        label.style['font-size'] = 14;
+        label.innerHTML = "Bits:";
+        div.append(label);
+
+        let bitSelect = document.createElement("select");
+        bitSelect.style['margin-top'] = 4;
+        bitSelect.style['margin-bottom'] = 4;
+        bitSelect.style['margin-left'] = 4;
+        div.append(bitSelect);
+
+        this.centerDiv.appendChild(div);
+        this.centerDiv.appendChild(document.createElement('br'));
+
+        // Populate the bitdepth selection
+        for (let bitdepth of [12, 10, 8, 7, 6, 5, 4, 3, 2, 1])
+        {
+            var opt = document.createElement("option");
+            opt.setAttribute('value', bitdepth);
+            opt.innerHTML = '' + bitdepth;
+            opt.selected = (bitdepth == state.params.bitdepth);
+            bitSelect.appendChild(opt);
+        }
```

Comment:
Here I'd like to have an option for 32 meaning no bit depth reduction.

I think it would also look nicer if the drop box read "8 bits" without a description on the left.